keras_cv/models/segmentation/deeplab_v3_plus/deeplab_v3_plus.py

```diff
@@ -229,16 +233,99 @@ def from_config(cls, config):
+        return copy.deepcopy({**backbone_presets, **deeplab_v3_plus_presets})
 
     @classproperty
     def presets_with_weights(cls):
         """Dictionary of preset names and configurations that include
         weights."""
-        return copy.deepcopy(backbone_presets_with_weights)
+        return copy.deepcopy(
+            {**backbone_presets_with_weights, **deeplab_v3_plus_presets}
+        )
 
     @classproperty
     def backbone_presets(cls):
         """Dictionary of preset names and configurations of compatible
         backbones."""
         return copy.deepcopy(backbone_presets)
+
+    @classmethod
+    def from_preset(
+        cls,
+        preset,
+        load_weights=None,
+        input_shape=None,
+        **kwargs,
+    ):
+        """Instantiate {{model_name}} model from preset config and weights.
+
+        Args:
+            preset: string. Must be one of "{{preset_names}}".
+                If looking for a preset with pretrained weights, choose one of
+                "{{preset_with_weights_names}}".
+            load_weights: Whether to load pre-trained weights into model.
+                Defaults to `None`, which follows whether the preset has
+                pretrained weights available.
+
+        Examples:
+        ```python
+        # Load architecture and weights from preset
+        model = keras_cv.models.{{model_name}}.from_preset(
+            "{{example_preset_name}}",
+        )
+
+        # Load randomly initialized model from preset architecture with weights
+        model = keras_cv.models.{{model_name}}.from_preset(
+            "{{example_preset_name}}",
+            load_weights=False,
+        ```
+        """
+        if not cls.presets:
+            raise NotImplementedError(
+                "No presets have been created for this class."
+            )
+
+        if preset not in cls.presets:
+            raise ValueError(
+                "`preset` must be one of "
+                f"""{", ".join(cls.presets)}. Received: {preset}."""
+            )
+
+        if load_weights and preset not in cls.presets_with_weights:
+            raise ValueError(
+                f"""Pretrained weights not available for preset "{preset}". """
+                "Set `load_weights=False` to use this preset or choose one of "
+                "the following presets with weights:"
+                f""" "{'", "'.join(cls.presets_with_weights)}"."""
+            )
+
+        metadata = cls.presets[preset]
+        # Check if preset is backbone-only model
+        if preset in cls.backbone_presets:
+            backbone_cls = keras.saving.get_registered_object(
+                metadata["class_name"]
+            )
+            backbone = backbone_cls.from_preset(preset, load_weights)
+            return cls(backbone, **kwargs)
+
+        # Otherwise must be one of class presets
+        config = metadata["config"]
+        config["backbone"]["config"]["input_shape"] = input_shape
```

Comment:
Sorry if I'm behind on this conversation -- can't we just do this in `Task` instead of just for DeepLab?

Reply:
we can! Just updated.